internal/api/courses/assignments/submissions/proxy/regrade.go

```diff
@@ -0,0 +1,60 @@
+package proxy
+
+import (
+	"github.com/edulinq/autograder/internal/api/core"
+	"github.com/edulinq/autograder/internal/db"
+	"github.com/edulinq/autograder/internal/grader"
+	"github.com/edulinq/autograder/internal/model"
+)
+
+type RegradeRequest struct {
+	core.APIRequestAssignmentContext
+	core.MinCourseRoleGrader
+
+	// Filter results to only users with this role.
+	FilterRole model.CourseUserRole `json:"filter-role"`
```

Comment:
I think we should target emails rather than roles.
If no emails, do all students (so, default to `"student"`).

I personally think that we should think about phasing out `filter-role` and go for how we handle email resolution instead.